File: src/1700-Number-of-Students-Unable-to-Eat-Lunch/number_of_students_unable_to_eat_lunch.py
from typing import List

class Solution:
    def countStudents(self, students: List[int], sandwiches: List[int]) -> int:
        check: int = 0
        while True:
            if sandwiches[0] == students[0]:
                students.pop(0)
                sandwiches.pop(0)
                check = 0
            else:
                temp: int = students.pop(0)
                students.append(temp)
                check += 1

            if len(sandwiches) == 0:
                break
            elif check == len(sandwiches):
                break

        return len(students)
    
if __name__ == "__main__":
    students: List[int] = [1,1,0,0]
    sandwiches: List[int] = [0,1,0,1]
    solution = Solution()
    
    print(solution.countStudents(students, sandwiches))

File: src/1700-Number-of-Students-Unable-to-Eat-Lunch/number_of_students_unable_to_eat_lunch.cpp
#include <iostream>
#include <vector>

class Solution {
public:
    int countStudents(std::vector<int>& students, std::vector<int>& sandwiches) {
        int check = 0;
        
        while (true) {
            if (sandwiches[0] == students[0]) {
                students.erase(students.begin());
                sandwiches.erase(sandwiches.begin());
                check = 0;
            } else {
                int temp = students[0];
                students.erase(students.begin());
                students.push_back(temp);
                check++;
            }

            if (sandwiches.empty()) {
                break;
            } else if (check == sandwiches.size()) {
                break;
            }
        }
        return students.size();
    }
};

int main() {
    std::vector<int> students = {1, 1, 0, 0};
    std::vector<int> sandwiches = {0, 1, 0, 1};
    Solution solution;

    std::cout << solution.countStudents(students, sandwiches);
    return 0;
}